### ISSUE TITLE
Race condition in max_gauge.go

### ISSUE THREAD

**Issue by dmitsh** [prometheus/prometheus#5643]
## Bug Report
**What did you do?**
I'm running tests for remote write API, and I'm seeing race condition
```
go run -race -tags netgo  ./cmd/prometheus/main.go --config.file prometheus.yml
```

**What did you see instead? Under which circumstances?**

**Environment**

* System information:

	Linux 4.15.0-51-generic x86_64

* Prometheus version:

	prometheus, version 2.10.0 (branch: upstream-master, revision: 819205ed0315069dfa4236925b3f2ec4615a8fb9)
	go version:       go1.12.4

* Prometheus configuration file:
```
# my global config
global:
  scrape_interval:     15s
  evaluation_interval: 30s
  # scrape_timeout is set to the global default (10s).

remote_write:
  - url: http://mydb/push

scrape_configs:
- job_name: test

  honor_labels: true
  # scrape_interval is defined by the configured global (15s).
  # scrape_timeout is defined by the global default (10s).

  # metrics_path defaults to '/metrics'
  # scheme defaults to 'http'.

  static_configs:
  - targets: ['localhost:12345']
```

* Logs:
```
==================
WARNING: DATA RACE
Write at 0x000004cc9248 by goroutine 43:
  github.com/prometheus/prometheus/storage/remote.(*maxGauge).Set()
      /home/vagrant/gopath/src/github.com/prometheus/prometheus/storage/remote/max_gauge.go:32 +0xb8
  github.com/prometheus/prometheus/storage/remote.(*timestampTracker).Commit()
      /home/vagrant/gopath/src/github.com/prometheus/prometheus/storage/remote/write.go:73 +0x153
  github.com/prometheus/prometheus/storage.(*fanoutAppender).Commit()
      /home/vagrant/gopath/src/github.com/prometheus/prometheus/storage/fanout.go:167 +0x481
  github.com/prometheus/prometheus/scrape.(*timeLimitAppender).Commit()
      <autogenerated>:1 +0x62
  github.com/prometheus/prometheus/scrape.(*scrapeLoop).append()
      /home/vagrant/gopath/src/github.com/prometheus/prometheus/scrape/scrape.go:1218 +0x105a
  github.com/prometheus/prometheus/scrape.(*scrapeLoop).run()
      /home/vagrant/gopath/src/github.com/prometheus/prometheus/scrape/scrape.go:935 +0xac6
```

**Comment by dmitsh** [comment]
I will post a PR soon.

**Comment by cstyan** [comment]
fixed in #5647
